packages/calcite-components/src/tests/stackedFocusTrap.e2e.ts

```diff
@@ -0,0 +1,130 @@
+      </div>
+      <calcite-button slot="back">Open Another Modal</calcite-button>
+    </calcite-modal>
+
+    <calcite-modal id="another-modal">
+      <div slot="content">
+        <p>
+          This is an example of a another modal that opens from a modal. This modal an input date picker, a popover and
+          a tooltip.
+        </p>
+        <calcite-popover placement="auto" reference-element="popover-button" closable id="popover">
+          <div>
+            <p>Example Popover.</p>
+            <calcite-label>
+              Input Date Picker
+              <calcite-input-date-picker id="input-date-picker"></calcite-input-date-picker>
+            </calcite-label>
+            <calcite-label>
+              Input Time Picker
+              <calcite-input-time-picker
+                name="calcite-input-time-picker"
+                id="input-time-picker"
+              ></calcite-input-time-picker>
+            </calcite-label>
+          </div>
+        </calcite-popover>
+        <calcite-button id="popover-button">Example Popover</calcite-button>
+        <calcite-button id="tooltip-auto-ref">auto</calcite-button>
+      </div>
+    </calcite-modal>
+  `;
+
+  it("closes a stack of open components sequentially in visual order", async () => {
+    const page = await newE2EPage();
+    await page.setContent(componentStack);
+    await skipAnimations(page);
+
+    async function testStackEscapeSequence(page: E2EPage, pickerType: string): Promise<void> {
+      async function openAndCheckVisibility(element: E2EElement): Promise<void> {
+        const elTagNameCamelCased = camelCase(element.tagName);
+        const openEvent = page.waitForEvent(`${elTagNameCamelCased}Open`);
+
+        element.setProperty("open", true);
+        await page.waitForChanges();
+        await openEvent;
+      }
+
+      async function testEscapeAndAssertOpenState(focusTrapOrderElements: E2EElement[]): Promise<void> {
+        for (let i = 0; i < focusTrapOrderElements.length; i++) {
+          const elTagNameCamelCased = camelCase(focusTrapOrderElements[i].tagName);
+          const closeEvent = page.waitForEvent(`${elTagNameCamelCased}Close`);
+
+          const activeElementId = await page.evaluate(() => document.activeElement?.id);
+
+          // 'input-time-picker', 'input-date-picker' retain focus after pressing esc
+          if (activeElementId) {
+            if (activeElementId === "input-time-picker") {
+              await page.keyboard.press("Tab");
+            } else if (activeElementId === "input-date-picker") {
+              await page.keyboard.down("Shift");
+              await page.keyboard.press("Tab");
+            }
+          }
+          await page.waitForChanges();
+
+          const activeElementIdAfterTab = await page.evaluate(() => document.activeElement?.id);
+
+          const expectedElementId =
+            focusTrapOrderElements[i].id === "sheet"
+              ? "sheet-button"
```

Comment:
This can be done in a follow-up refactor PR, but it'd be helpful to add a comment explaining why we check for `sheet-button` when `sheet` is the next element in the list.

Reply:
I'll add it now.